### ISSUE TITLE
VB does not produce an error when Conditional is placed on Function members

### ISSUE THREAD

**Issue by jhinder** [dotnet/roslyn#32660]
**Version Used**: VS 16.0 Preview 1.1

**Steps to Reproduce**:

1. Copy the code below into a VB project.
2. Compile.

```vbnet
Public Class ConditionalAttrRepro
    Public ReadOnly Property I As Integer
        <Conditional("SOMETHING")>
        Get
            Return 0
        End Get
    End Property

    <Conditional("SOMETHING")>
    Public Function M() As Integer
        Return 0
    End Function
End Class
```

**Expected Behavior**: The code does not compile, because the Conditional attribute is placed on invalid members.

From the [docs](https://docs.microsoft.com/en-gb/dotnet/api/system.diagnostics.conditionalattribute?view=netframework-4.7.2#remarks):

> You will get a compilation error in Visual Studio if you apply this attribute to a method that does not return void.

**Actual Behavior**: The code compiles with warning BC41007 ("Attribute 'Conditional' is only valid on 'Sub' declarations"). No idea what would happen if a compiler consuming the assembly were to respect the attribute.

**Comment by gafter** [comment]
That warning should probably be an error, but it may be too late to change it.
